src/components/Facility/AssetCreate.tsx

```diff
@@ -510,400 +515,398 @@ const AssetCreate = (props: AssetProps) => {
           </div>
           <div className="flex h-full w-full overflow-auto xl:ml-72">
             <div className="w-full max-w-3xl 2xl:max-w-4xl">
-              <form
-                onSubmit={(e) => handleSubmit(e, false)}
+              <Form<AssetData>
+                disabled={isLoading}
+                defaults={initAssetData}
```

Comment:
Switch to using `asyncGetDefaults` instead. This allows asynchronously fetching initial data and simplifies lot of the state management.

Reply:
current code uses 
const assetQuery = useQuery(routes.getAsset, {
    pathParams: { external_id: assetId! },
    prefetch: !!assetId,
which is not async?

Reply:
you can disable prefetch and call refetch in asyncGetDefaults